tests/test_augmentation.py

```diff
@@ -36,9 +36,55 @@ def test_channel_swap_correctness(n_ch, data_format, data_type):
     #     kapre_augs.append(model(batch_src, training=True))
 
 
+@pytest.mark.parametrize('data_format', ['default', 'channels_first', 'channels_last'])
+def test_spec_augment_shape_correctness(data_format):
+    """
+    Checks that shapes are correct depending on each data_format
+    """
+
+    batch_src, input_shape = get_spectrogram(data_format)
+
+    model = tf.keras.Sequential()
+    spec_augment = SpecAugment(
+                               input_shape=input_shape,
+                               freq_mask_param=5,
+                               time_mask_param=10,
+                               n_freq_masks=4,
+                               n_time_masks=3,
+                               mask_value=0.,
+                               data_format=data_format)
+
+    model.add(spec_augment)
+
+    # We must force training to True to test properly if SpecAugment works as expected
+    spec_augmented = model(batch_src, training=True)[0]
+
```

Comment:
i get that it's difficult to test the behavior of the whole layer. but can't there be any way at all..?
also, at least we can test the static methods we defined.

Reply:
Sure, you mean the method for generating masks, applyng masks and similar, right? I

Reply:
yup, that's right.